### PR TITLE
Added Debian sid (#3)

### PR DESCRIPTION
**Pull request by moul**

### CODE REVIEW

File: sid/Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+DOCKER_NAMESPACE =	armbuild/
+NAME =			scw-distrib-debian
+VERSION =		sid
+VERSION_ALIASES =	
+TITLE =			Debian Sid
+DESCRIPTION =		Debian Sid/Unstable
+SOURCE_URL =		https://github.com/scaleway/image-debian
+VENDOR_URL =		https://wiki.debian.org/DebianUnstable
+
+IMAGE_NAME =		Debian Sid/Unstable
```
Comment (QuentinPerez) [inline sid/Makefile]: "Debian Sid/Unstable", When you want to build this image with `make image`, It will crash

File: sid/patches/etc/locale.gen [hunk 2]
```diff
@@ -0,0 +1,2 @@
+en_US.UTF-8 UTF-8
```
Comment (QuentinPerez) [inline sid/patches/etc/locale.gen]: This file will be erased with `RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && locale-gen` in the Dockerfile

File: sid/patches/etc/update-motd.d/50-scw [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+if [ -z "$DISTRIB_DESCRIPTION" ] && [ -x /usr/bin/lsb_release ]; then
+	# Fall back to using the very slow lsb_release utility
+	DISTRIB_DESCRIPTION=$(lsb_release -s -d)
+fi
+
+[ -r /etc/scw-release ] && . /etc/scw-release
+
+METADATA=$(oc-metadata --cached)
+
+metadata() {
```
Comment (QuentinPerez) [inline sid/patches/etc/update-motd.d/50-scw]: Use `oc-metadata --cached YOUR_VARIABLE` instead of calling a function

File: sid/Makefile (hunk 1 above)
Comment (moul) [inline sid/Makefile]: It should not, IMAGE_NAME should only be used for `scw tag` which can be easily escaped

File: sid/patches/etc/locale.gen (hunk 2 above)
Comment (moul) [inline sid/patches/etc/locale.gen]: yes, I will remove it

File: sid/Makefile (hunk 1 above)
Comment (QuentinPerez) [inline sid/Makefile]: An example with `ELK`

`chmod +x /tmp/create-image-from-http.sh
VOLUME_SIZE=50G IMAGE_NAME=ELK stack IMAGE_BOOTSCRIPT=stable /tmp/create-image-from-http.sh http://212.47.249.205/scw-app-elk-latest/rootfs.tar
/bin/sh: 1: stack: not found
docker-rules.mk:106: recipe for target 'image_on_local' failed`

We can also added `"` in the `docker-rules.mk`

File: sid/patches/etc/locale.gen (hunk 2 above)
Comment (moul) [inline sid/patches/etc/locale.gen]: I kept this file and remove the dummy `/etc/locale.gen` creation from the Dockerfile

File: sid/Makefile (hunk 1 above)
Comment (moul) [inline sid/Makefile]: Yes we should :)

Comment (moul) [inline sid/Makefile]: I will fix the script and open a new PR on image-tools

**Comment by moul** [comment]
```console
root@docker-builder-6:~/image-debian/sid# make image_on_store
test -f /tmp/create-image-from-http.sh \
        || wget -qO /tmp/create-image-from-http.sh https://github.com/scaleway/scaleway-cli/raw/master/examples/create-image-from-http.sh
chmod +x /tmp/create-image-from-http.sh
rsync -Pave ssh /tmp/build/scw-distrib-debian-sid/rootfs.tar store.scw.42.am:store/scw/scw-distrib-debian-sid.tar
Enter passphrase for key '/root/.ssh/id_rsa':
sending incremental file list

sent 50 bytes  received 12 bytes  17.71 bytes/sec
total size is 660,449,280  speedup is 10,652,407.74
http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
VOLUME_SIZE="50G" IMAGE_NAME="Debian Sid/Unstable" IMAGE_BOOTSCRIPT="latest" /tmp/create-image-from-http.sh http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
[+] URL of the tarball: http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
[+] Target name: scw-distrib-debian-sid.tar
[+] Creating new server in rescue mode with a secondary volume...
/tmp/create-image-from-http.sh: line 29: scw: command not found
../docker-rules.mk:101: recipe for target 'image_on_store' failed
make: *** [image_on_store] Error 127
root@docker-builder-6:~/image-debian/sid# ^C
root@docker-builder-6:~/image-debian/sid# ^C
root@docker-builder-6:~/image-debian/sid# clear; make image_on_store
test -f /tmp/create-image-from-http.sh \
        || wget -qO /tmp/create-image-from-http.sh https://github.com/scaleway/scaleway-cli/raw/master/examples/create-image-from-http.sh
chmod +x /tmp/create-image-from-http.sh
rsync -Pave ssh /tmp/build/scw-distrib-debian-sid/rootfs.tar store.scw.42.am:store/scw/scw-distrib-debian-sid.tar
Enter passphrase for key '/root/.ssh/id_rsa':
sending incremental file list

sent 50 bytes  received 12 bytes  24.80 bytes/sec
total size is 660,449,280  speedup is 10,652,407.74
http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
VOLUME_SIZE="50G" IMAGE_NAME="Debian Sid/Unstable" IMAGE_BOOTSCRIPT="latest" /tmp/create-image-from-http.sh http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
[+] URL of the tarball: http://store.scw.42.am/scw/scw-distrib-debian-sid.tar
[+] Target name: scw-distrib-debian-sid.tar
[+] Creating new server in rescue mode with a secondary volume...
[+] Server created: 00000000-0000-1000-0000-000000000000
[+] Booting...
Enter passphrase for key '/root/.ssh/id_rsa':
Linux image-writer-scw-distrib-debian-sid-tar 3.2.34-30 #17 SMP Mon Apr 13 15:53:45 UTC 2015 armv7l armv7l armv7l GNU/Linux
[+] Server is booted
[+] Formating and mounting /dev/nbd1...
Enter passphrase for key '/root/.ssh/id_rsa':
mke2fs 1.42.9 (4-Feb-2014)
Filesystem label=
OS type: Linux
Block size=4096 (log=2)
Fragment size=4096 (log=2)
Stride=0 blocks, Stripe width=0 blocks
3055616 inodes, 12207031 blocks
610351 blocks (5.00%) reserved for the super user
First data block=0
Maximum filesystem blocks=0
373 block groups
32768 blocks per group, 32768 fragments per group
8192 inodes per group
Superblock backups stored on blocks:
        32768, 98304, 163840, 229376, 294912, 819200, 884736, 1605632, 2654208,
        4096000, 7962624, 11239424

Allocating group tables: done
Writing inode tables: done
Creating journal (32768 blocks): done
Writing superblocks and filesystem accounting information: done

[+] /dev/nbd1 formatted in ext4 and mounted on /mnt
[+] Download tarball from S3 and write it to /dev/nbd1
Enter passphrase for key '/root/.ssh/id_rsa':
Enter passphrase for key '/root/.ssh/id_rsa':
[+] Tarball extracted on /dev/nbd1
[+] Stopping the server
[+] Server stopped
[+] Creating a snapshot of nbd1
[+] Snapshot 11111111-1111-1111-1111-111111111111 created
[+] Creating an image based of the snapshot
[+] Image created: 22222222-2222-1222-2222-222222222222
[+] Deleting temporary server
[+] Server deleted
```

---

```console
$ scw inspect 00000000-0000-1000-0000-000000000000
```
```json
[{
  "id": "00000000-0000-1000-0000-000000000000",
  "name": "Debian Sid/Unstable",
  "creation_date": "2015-09-11T11:30:41.842288+00:00",
  "modification_date": "2015-09-11T11:30:41.842288+00:00",
  "root_volume": {
    "id": "11111111-1111-1111-1111-111111111111",
    "size": 50000000000,
    "name": "scw-distrib-debian-sid-2015-09-11_11:26",
    "volume_type": "l_ssd"
  },
  "default_bootscript": {
    "id": "22222222-2222-1222-2222-222222222222",
    "title": "4.1.6-std #249 (latest)",
    "bootcmdargs": {
      "id": "33333333-3333-1333-3333-333333333333",
      "value": "LINUX_COMMON ip=dhcp boot=local root=/dev/nbd0 nbd.max_parts=8"
    },
    "initrd": {
      "id": "44444444-4444-1444-4444-444444444444",
      "path": "initrd/c1-uInitrd-v3.3",
      "title": "C1 uInitrd v3.3"
    },
    "kernel": {
      "id": "55555555-5555-1555-5555-555555555555",
      "dtb": "dtb/pimouss-computing.dtb.4.1.6-249-std",
      "path": "kernel/pimouss-uImage-4.1.6-249-std",
      "title": "Pimouss 4.1.6-249-std"
    },
    "public": true
  },
  "organization": "66666666-6666-1666-6666-666666666666",
  "arch": "arm"
}]
```

**Comment by moul** [comment]
```
Welcome to Scaleway' initrd            http://j.mp/scw-initrd-linux

███████╗ ██████╗ █████╗ ██╗     ███████╗██╗    ██╗ █████╗ ██╗   ██╗
██╔════╝██╔════╝██╔══██╗██║     ██╔════╝██║    ██║██╔══██╗╚██╗ ██╔╝
███████╗██║     ███████║██║     █████╗  ██║ █╗ ██║███████║ ╚████╔╝
╚════██║██║     ██╔══██║██║     ██╔══╝  ██║███╗██║██╔══██║  ╚██╔╝
███████║╚██████╗██║  ██║███████╗███████╗╚███╔███╔╝██║  ██║   ██║
╚══════╝ ╚═════╝╚═╝  ╚═╝╚══════╝╚══════╝ ╚══╝╚══╝ ╚═╝  ╚═╝   ╚═╝

>>> Initrd build: v3.3.0-0-g6c531b5 - 2015/09/07
>>> Kernel version: 4.1.6-249

>>> Installing busybox...
>>> Cloning initramfs for shutdown......
>>> Mounting pseudo-filesystems......
>>> Mounting /dev...
>>> Mounting /dev/pts...
>>> Mounting /run...
>>> Mounting /sys...
>>> Mounting /proc...

>>> Kernel cmdline: earlyprintk=ttyS0 console=ttyS0,9600n8 nousb noplymouth ip=dhcp boot=local root=/dev/nbd0 nbd.max_parts=8

>>> Configuring GPIOs...
>>> Configuring network...

>>> Server ID: 3dcf2da8-9afb-40f9-af51-979cfe68039e
>>> Private IP: 10.1.12.25
>>> Public IP: 212.47.230.118
>>> Hostname: serene-blackwell

>>> Checking user input for debug/verbose mode...
>>> Checking user input for debug shell...
>>> Signaling the server is kernel-started to the control plane...
>>> Adjusting time (ntp)...
>>> Initializing 'local' root file system...
>>> Attaching nbd0...
>>> Mounting nbd0 root...
>>> Mounting /newroot...
>>> Checking metadata in case we skip to wait for secondary volumes to be ready...
>>> Signaling the server is booted to the control plane...
>>> Creating default configuration files on rootfs......
>>> Checking metadata for debug shell...
>>> Checking metadata for debug sshd (dropbear)...
>>> Changing root password...
>>> Mounting /newroot/proc...
>>> Mounting /newroot/sys...
>>> Mounting /newroot/run...
>>> Mounting /newroot/dev...
>>> Moving initramfs backup to /run/initramfs for shutdown...
>>> Switching to linux...
End of Scaleway' initrd
           _ _       _      _          _ _
 ___ _ _ _|_| |_ ___| |_   | |_ ___   | |_|___ _ _ _ _
|_ -| | | | |  _|  _|   |  |  _| . |  | | |   | | |_'_|
|___|_____|_|_| |___|_|_|  |_| |___|  |_|_|_|_|___|_,_|


Welcome to Debian GNU/Linux stretch/sid!

[  OK  ] Created slice User and Session Slice.
[  OK  ] Reached target Encrypted Volumes.
[  OK  ] Reached target Remote File Systems (Pre).
[  OK  ] Reached target Swap.
[  OK  ] Created slice System Slice.
[  OK  ] Created slice system-serial\x2dgetty.slice.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Started Forward Password Requests to Wall Directory Watch.
         Starting Increase datagram queue length...
[  OK  ] Listening on udev Control Socket.
[  OK  ] Created slice system-getty.slice.
[  OK  ] Started Dispatch Password Requests to Console Directory Watch.
[  OK  ] Reached target Paths.
[  OK  ] Reached target Network is Online.
[  OK  ] Reached target Slices.
[  OK  ] Listening on Journal Socket.
         Starting Load Kernel Modules...
         Starting Create Static Device Nodes in /dev...
         Mounting POSIX Message Queue File System...
         Starting Remount Root and Kernel File Systems...
         Starting SCW kernel requirements checker...
         Mounting Debug File System...
[  OK  ] Listening on udev Kernel Socket.
[  OK  ] Listening on /dev/initctl Compatibility Named Pipe.
[UNSUPP] Starting of Arbitrary Executable Fi...m Automount Point not supported.
[  OK  ] Mounted POSIX Message Queue File System.
[  OK  ] Started Load Kernel Modules.
[  OK  ] Started Remount Root and Kernel File Systems.
[  OK  ] Mounted Debug File System.
[  OK  ] Started Create Static Device Nodes in /dev.
[  OK  ] Started Increase datagram queue length.
[  OK  ] Started SCW kernel requirements checker.
[  OK  ] Listening on Syslog Socket.
         Starting Journal Service...
         Starting udev Kernel Device Manager...
[  OK  ] Reached target Local File Systems (Pre).
         Starting Load/Save Random Seed...
[  OK  ] Reached target Local File Systems.
[  OK  ] Reached target Remote File Systems.
         Starting udev Coldplug all Devices...
         Starting Apply Kernel Variables...
[  OK  ] Started Load/Save Random Seed.
[  OK  ] Started Apply Kernel Variables.
[  OK  ] Started Journal Service.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started Flush Journal to Persistent Storage.
[  OK  ] Started udev Kernel Device Manager.
         Starting Create Volatile Files and Directories...
[  OK  ] Started Create Volatile Files and Directories.
         Starting Update UTMP about System Boot/Shutdown...
[  OK  ] Reached target System Time Synchronized.
[  OK  ]
Debian GNU/Linux stretch/sid serene-blackwell ttyS0

serene-blackwell login:
```

Only 1 warning for the boot

**Comment by moul** [comment]
```console
$ scw run f78f4a1b-7fd3-4ad3-b392-dd45aee4854e

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
-bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
-bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
root@desperate-chandrasekhar:~#
```

* update-motd.t not working
* problem with locales

**Comment by moul** [comment]
```console
root@desperate-chandrasekhar:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux stable-updates (sid)
Release:	stable-updates
Codename:	sid
```

* lsb_release: OK

**Comment by moul** [comment]
```console
root@desperate-chandrasekhar:~# apt-get update
Ign http://ftp.fr.debian.org stable InRelease
Hit http://ftp.fr.debian.org jessie-updates InRelease
Hit http://ftp.fr.debian.org testing InRelease
Hit http://ftp.fr.debian.org sid InRelease
Hit http://security.debian.org stable/updates InRelease
Hit http://security.debian.org testing/updates InRelease
Hit http://ftp.fr.debian.org stable Release.gpg
Hit http://ftp.fr.debian.org stable Release
Get:1 http://ftp.fr.debian.org jessie-updates/main armhf Packages/DiffIndex [643 B]
Get:2 http://ftp.fr.debian.org jessie-updates/main Translation-en/DiffIndex [229 B]
Hit http://ftp.fr.debian.org jessie-updates/non-free armhf Packages
Hit http://ftp.fr.debian.org jessie-updates/contrib armhf Packages
Hit http://ftp.fr.debian.org jessie-updates/contrib Translation-en
Hit http://ftp.fr.debian.org jessie-updates/non-free Translation-en
Get:3 http://ftp.fr.debian.org testing/main armhf Packages/DiffIndex [7876 B]
Get:4 http://ftp.fr.debian.org testing/contrib armhf Packages/DiffIndex [7819 B]
Get:5 http://ftp.fr.debian.org testing/non-free armhf Packages/DiffIndex [7819 B]
Get:6 http://ftp.fr.debian.org testing/contrib Translation-en/DiffIndex [4921 B]
Get:7 http://ftp.fr.debian.org testing/main Translation-en/DiffIndex [7876 B]
Get:8 http://ftp.fr.debian.org testing/non-free Translation-en/DiffIndex [6163 B]
Get:9 http://ftp.fr.debian.org sid/contrib Translation-en/DiffIndex [7819 B]
Get:10 http://ftp.fr.debian.org sid/main Translation-en [5068 kB]
Hit http://security.debian.org stable/updates/main armhf Packages
Hit http://security.debian.org stable/updates/contrib armhf Packages
Hit http://security.debian.org stable/updates/non-free armhf Packages
Hit http://security.debian.org stable/updates/contrib Translation-en
Hit http://security.debian.org stable/updates/main Translation-en
Hit http://security.debian.org stable/updates/non-free Translation-en
Hit http://security.debian.org testing/updates/main armhf Packages
Hit http://security.debian.org testing/updates/contrib armhf Packages
Hit http://security.debian.org testing/updates/non-free armhf Packages
Hit http://security.debian.org testing/updates/contrib Translation-en
Hit http://security.debian.org testing/updates/main Translation-en
Hit http://security.debian.org testing/updates/non-free Translation-en
Get:11 http://ftp.fr.debian.org sid/non-free Translation-en [80.5 kB]
Get:12 http://ftp.fr.debian.org sid/main armhf Packages [7404 kB]
Get:13 http://ftp.fr.debian.org sid/contrib armhf Packages [46.2 kB]
Get:14 http://ftp.fr.debian.org sid/non-free armhf Packages [69.4 kB]
Get:15 http://ftp.fr.debian.org stable/main armhf Packages [6620 kB]
Get:16 http://ftp.fr.debian.org stable/contrib armhf Packages [38.1 kB]
Get:17 http://ftp.fr.debian.org stable/non-free armhf Packages [62.2 kB]
Get:18 http://ftp.fr.debian.org stable/contrib Translation-en [38.4 kB]
Get:19 http://ftp.fr.debian.org stable/main Translation-en [4585 kB]
Get:20 http://ftp.fr.debian.org stable/non-free Translation-en [72.2 kB]
Fetched 24.1 MB in 20s (1188 kB/s)
Reading package lists... Done
root@desperate-chandrasekhar:~# apt-get upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages will be upgraded:
  sysstat
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 293 kB of archives.
After this operation, 1024 B of additional disk space will be used.
Do you want to continue? [Y/n]
Get:1 http://ftp.fr.debian.org/debian/ sid/main sysstat armhf 11.1.6-2 [293 kB]
Fetched 293 kB in 0s (4109 kB/s)
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = "en_US.UTF-8",
        LC_CTYPE = "UTF-8",
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 26746 files and directories currently installed.)
Preparing to unpack .../sysstat_11.1.6-2_armhf.deb ...
Unpacking sysstat (11.1.6-2) over (11.1.6-1) ...
Processing triggers for systemd (226-1) ...
Processing triggers for man-db (2.7.3-1) ...
Setting up sysstat (11.1.6-2) ...
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
root@desperate-chandrasekhar:~#
```

* `apt-get update && apt-get upgrade`: OK

**Comment by moul** [comment]
* Reboot: OK

**Comment by moul** [comment]
Locale fixed, but still no dynamic motd

Not blocking for a first release

**Comment by moul** [comment]
LGTM

**Comment by QuentinPerez** [comment]
LGTM

**Comment by aimxhaisse** [comment]
LGTM

**Pull request merged**